### PR TITLE
 Setting up configuration for eslint, husky, prettier

### LINKED ISSUE #12: Set up Eslint, Husky
Need to setup configuration files for 

ESlint :- It is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code. In many ways, it is similar to JSLint and JSHint with a few exceptions

Husky with lint-staged : For pre-commit hookws

### PR DESCRIPTION
**Pull request by Udit-takkar**
Closes #12

### CODE REVIEW

**Comment by Udit-takkar** [comment]
currently setting up eslint config and rules airbnb

**Comment by Udit-takkar** [comment]
I update some prettier config to use singleQuote, tabWidth:2 etc
Eslint: I have set up eslint with airbnb style to enforce good coding style and structures. Had to add new parser `@babel/eslint-parser`  the old one depreceated and one preset `@babel/preset-react`.
Husky and lint-staged: which would run pre-commit hook to lint and format whatever code you commit.

**Review by sohamsshah — COMMENTED** [review]
Hey @Udit-takkar , the implementation looks good to me! 
I would suggest that since we have newer commands added in `package.json`, it would be great to update README.md so that contributors lint their code when pushing.

**Comment by Udit-takkar** [comment]
@sohamsshah  Okay I would once this get merged

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-    "github.copilot.autocomplete.enable": true,
-    "github.copilot.list.count": 5
-}
\ No newline at end of file
+  "github.copilot.autocomplete.enable": true,
+  "github.copilot.list.count": 5
+}
```
Comment (YashKumarVerma) [inline .vscode/settings.json]: not required on the project repo

File: package.json [hunk 2]
```diff
@@ -39,8 +51,24 @@
...
   },
   "devDependencies": {
+    "@babel/eslint-parser": "^7.16.3",
+    "@babel/preset-react": "^7.16.0",
     "autoprefixer": "^9.8.6",
+    "eslint": "^8.4.1",
+    "eslint-config-airbnb": "^19.0.2",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-config-react-app": "^6.0.0",
+    "eslint-plugin-flowtype": "^8.0.3",
```
Comment (YashKumarVerma) [inline package.json]: are we using flow types in the codebase?

File: src/App.test.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
-});
+// import App from './App';
+
+// test('renders learn react link', () => {
+//   render(<App />);
+//   const linkElement = screen.getByText(/learn react/i);
+//   expect(linkElement).toBeInTheDocument();
+// });
```
Comment (YashKumarVerma) [inline src/App.test.js:9]: any particular reason to comment them out?

File: .prettierrc [hunk 4]
```diff
@@ -1,3 +1,7 @@
 {
-  "arrowParens": "always"
+  "printWidth": 80,
+  "singleQuote": true,
+  "trailingComma": "all",
+  "tabWidth": 2,
+  "arrowParens": "avoid"
```
Comment (YashKumarVerma) [inline .prettierrc]: always put parenthesis

**Review by YashKumarVerma — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (Udit-takkar) [reply .vscode/settings.json]: ok

File: src/App.test.js (hunk 3 above)
Comment (Udit-takkar) [reply src/App.test.js:9]: there were not being used

File: .prettierrc (hunk 4 above)
Comment (Udit-takkar) [reply .prettierrc]: Ok I'll change this

File: package.json (hunk 2 above)
Comment (Udit-takkar) [reply package.json]: no i think i did it by mistake

**Pull request closed without merging**